### ISSUE TITLE
Warn about wrong Object.defineProperty usage

### ISSUE THREAD

**Issue by BridgeAR** [eslint/eslint#10315]
**Please describe what the rule should do:**

The rule should warn about faulty uses of `Object.defineProperty` and `Object.defineProperties`. It is not possible to combine a `value` property with a getter or setter or to combine the `writable` property with a getter or setter.

**What category of rule is this? (place an "X" next to just one item)**

[ ] Enforces code style
[x] Warns about a potential error
[ ] Suggests an alternate way of doing something
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
'use strict';

const obj = {};

Object.defineProperty(obj, 'foo', {
  writable: true,
  get() {}
});

Object.defineProperty(obj, 'foo', {
  value: 'foo',
  get() {}
});
```

**Why should this rule be included in ESLint (instead of a plugin)?**

It provides an early feedback for a mistake. Especially if code has no tests, this might cause issues in some applications where an error is thrown at runtime.

**Comment by platinumazure** [comment]
Hi @BridgeAR, thanks for the issue.

I agree with this. Not yet sure how best to do this (new rule makes sense but maybe this could fit with an existing?), hopefully others on the team may have some insights there.

**Comment by BridgeAR** [comment]
I have no strong opinion about adding a new rule for it or adding it to an existing one. Out of my head I am just not aware of any rule where it would fit in.

**Comment by ljharb** [comment]
This should also then warn about the second argument to Object.create?

**Comment by g-plane** [comment]
How about just checking the keys of the descriptor?
Like this: (just an idea for prototyping)
```javascript
module.exports = {
  create(context) {
    return {
      CallExpression(node) {
        const descriptor = node.arguments[2]
        if (descriptor.type === 'ObjectExpression') {
          const keys = descriptor.properties.map(prop => prop.key.name)
          if (keys.includes('writable') && keys.includes('get')) {
            context.report({
              node: descriptor,
              message: 'Invalid object descriptor'
            })
          }
        }
      }
    }
  }
}
```

And the result is:
![image](https://user-images.githubusercontent.com/17216317/39964824-a118470e-56bf-11e8-9535-77a6545d0c68.png)
